migration.rst

```diff
@@ -0,0 +1,415 @@
+                '_controller' => 'App\Controller\LegacyController::loadLegacyScript',
+                'requestPath' => '/' . $legacyScriptFile->getRelativePathname(),
+                'legacyScript' => $legacyScriptFile->getPathname(),
+            ]));
+        }
+
+        return $collection;
+    }
+
+This only shows the ``load``-method of a custom Route Loader. You will also
+have to register the loader in your application's ``routing.yaml`` as
+described in the documentation for :doc:`Custom Route Loaders </routing/custom_route_loader>`.
+Depending on your configuration you might also have to tag the service with
+``routing.loader``. Afterwards you should be able to see all the legacy routes
+in your route configuration, e.g. when you call the ``debug:router``-command:
+
+.. code-block:: terminal
+
+    $ php bin/console debug:router
+
+In order to use these routes you will need to create a controller that handles
+these routes. You might have noticed the ``_controller`` attribute we attached
+to our routes, which tells Symfony which Controller to call whenever we try
+to access one of our legacy routes. The controller itself can then use the
+attributes we passed to determine which script to call and wrap the output in
+a response class::
+
+    class LegacyController
+    {
+        public function loadLegacyScript($requestPath, $legacyScript)
+        {
+            return StreamedResponse::create(
+                function () use ($requestPath, $legacyScript) {
+                    $_SERVER['PHP_SELF'] = $requestPath;
+                    $_SERVER['SCRIPT_NAME'] = $requestPath;
+                    $_SERVER['SCRIPT_FILENAME'] = $legacyScript;
+
+                    chdir(dirname($legacyScript));
+
+                    require $legacyScript;
+                }
+            );
+        }
+    }
+
+This controller will set some server variables that might be needed by
+the legacy application, switches to the script directory to ensure all
+operations assuming this is the current working dir work properly and then
+requires the old script, which essentially calls the original script inside
+our current application scope.
+
+There are some risks to this approach, but since the legacy code now runs
+inside a controller action we gain access to many functionalities from the
+new Symfony application, including the chance to use Symfony's event
+lifecycle. For instance, this allows us to transition the authentication
+and authorization over to the Symfony application using the Security
+component and its firewalls.
+
+Additional Resources
+--------------------
+
+The topic of migrating from an existing application towards Symfony is
+sometimes discussed during conferences. For example the talk
+`Modernizing with Symfony`_ from SymfonyCon Berlin 2016 reiterates some
+of the points from this page.
+
+.. TODO: Add more resources like blog articles or conference talks
```

Comment:
I wouldn't add more references for now. Also, we're very picky about linking external resources from Symfony Docs because it's hard to ensure the quality of those resources.